quarkus-workshop-super-heroes/super-heroes/rest-narration/src/main/java/io/quarkus/workshop/superheroes/narration/SemanticKernelNarrationService.java

```diff
@@ -0,0 +1,93 @@
+import com.microsoft.semantickernel.textcompletion.CompletionSKFunction;
+import com.microsoft.semantickernel.textcompletion.TextCompletion;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.eclipse.microprofile.faulttolerance.Fallback;
+import org.eclipse.microprofile.faulttolerance.Timeout;
+import org.jboss.logging.Logger;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Properties;
+
+@ApplicationScoped
+public class SemanticKernelNarrationService implements NarrationService {
+
+    @Inject
+    Logger logger;
+
+    @Override
+    @Fallback(fallbackMethod = "fallbackNarrate")
+    @Timeout(30_000)
+    public String narrate(Fight fight) throws Exception {
+
+        // Creates an Azure OpenAI client
+        OpenAIAsyncClient client = getClient();
+        // OpenAIAsyncClient client = OpenAIClientProvider.getClient();
+
+        // Creates an instance of the TextCompletion service
+        TextCompletion textCompletion = SKBuilders.chatCompletion().withOpenAIClient(client).setModelId("deploy-semantic-kernel").build();
+
+        // Instantiates the Kernel
+        Kernel kernel = SKBuilders.kernel().withDefaultAIService(textCompletion).build();
+
+        // Registers skills
+        ReadOnlyFunctionCollection skill = kernel.importSkillFromDirectory("NarrationSkill", "src/main/resources", "NarrationSkill");
+        CompletionSKFunction fightFunction = skill.getFunction("NarrateFight", CompletionSKFunction.class);
+
+        // Ask for a Joke
+        SKContext fightContext = SKBuilders.context().build();
+        fightContext.setVariable("winner_team", fight.winnerTeam);
+        fightContext.setVariable("winner_name", fight.winnerName);
+        fightContext.setVariable("winner_powers", fight.winnerPowers);
+        fightContext.setVariable("winner_level", String.valueOf(fight.winnerLevel));
+        fightContext.setVariable("loser_team", fight.loserTeam);
+        fightContext.setVariable("loser_name", fight.loserName);
+        fightContext.setVariable("loser_powers", fight.loserPowers);
+        fightContext.setVariable("loser_level", String.valueOf(fight.loserLevel));
+        Mono<SKContext> result = fightFunction.invokeAsync(fightContext);
+
+        String narration = result.block().getResult();
+        logger.info("The narration for the fight is: " + narration);
+
+        return narration;
+    }
+
+    public String fallbackNarrate(Fight fight) {
+        logger.warn("Falling back on Narration");
+        return """
+            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
+            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
```

Comment:
Ok, *here* is where I add the comment about the fallback being more 'useful'. I think the aim of the fallback shouldn't be simply to prevent a nasty exception and/or stack trace; it should be to provide a graceful degradation. Low-quality text about how there was a fight is degraded, but graceful (IMO). Lorem ipsum is degraded, and not-graceful. It's a user visible error, so it's not that different from having a big red X in the UI.